extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -93,13 +93,19 @@ public final class HibernateReactiveProcessor {
     void registerBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans, CombinedIndexBuildItem combinedIndex,
             List<PersistenceUnitDescriptorBuildItem> descriptors,
             JpaModelBuildItem jpaModel) {
+        if (!hasEntities(jpaModel)) {
+            LOG.warnf("Skip registration of the %s bean - no JPA entities were found",
+                    ReactiveSessionFactoryProducer.class.getSimpleName());
+            return;
+        }
         if (descriptors.size() == 1) {
```

Comment:
I have to confess I'm not sure how you get to zero descriptor. But if this is what happens with unconfigured apps in DEV mode even though there's a DEV service for the DB, I think we should not throw.

Reply:
> I have to confess I'm not sure how you get to zero descriptor.

In the main branch it's empty if no `PersistenceUnitDescriptorBuildItem`  is produced, i.e. if no db kind is configured: https://github.com/quarkusio/quarkus/blob/main/extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java#L175-L197

> But if this is what happens with unconfigured apps in DEV mode even though there's a DEV service for the DB, I think we should not throw.

If we don't throw then we're in the same illegal situation as before - no `Mutiny.SessionFactory` bean available.

Reply:
In theory, multiple descriptors can be produced so I believe that it's necessary...